src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -128,7 +128,7 @@ private async Task ShowHotspotAsync(IShowHotspotRequest request)
 
         private void ShowFailureInfoBar()
         {
-            RunOnUIThread.Run(() => failureInfoBar.Show());
+            RunOnUIThread.Run(() => failureInfoBar.Show(HotspotsToolWindow.ToolWindowId));
```

Comment:
I saw the issue regarding the Clear.
I think that maybe the failureInfoBar should return a task and switch to the UI thread itself, instead of relying on the callers to do it.

Reply:
Yep; as I said in one of the PRs yesterday, I think we should consider making any new service that consumes VS services async.

As it happens, the failure info bar doesn't directly consume any VS services - it only consumes our services, so ideally our existing info bar manager should handle the threading requirements. I'll have a look, but that might involve too much change.